language_bindings/rust/rust_build/src/lib.rs

```diff
@@ -0,0 +1,360 @@
+            result = self.rename_runtime();
+            if result.is_ok() {
+                println!("cargo:rustc-link-lib=static={}", "runtime");
+            }
+        } else {
+            let errors =
+                String::from("failed to make runtime with gen: ").add(self.gen_name.as_str());
+            result = Err(io::Error::new(ErrorKind::Other, errors.as_str()));
+        }
+        result
+    }
+
+    fn rename_runtime(&self) -> Result<()> {
+        println!(
+            "file name: {:?}",
+            self.rs_output
+                .join("runtime")
+                .with_extension("a")
+                .to_str()
+                .unwrap()
+        );
+        println!(
+            "file out: {:?}",
+            self.rs_output
+                .join(String::from("lib").add("runtime"))
+                .with_extension("a")
+                .to_str()
+                .unwrap()
+        );
+        fs::rename(
+            self.rs_output
+                .join("runtime")
+                .with_extension("a")
+                .to_str()
+                .unwrap(),
+            self.rs_output
+                .join(String::from("lib").add("runtime"))
+                .with_extension("a")
+                .to_str()
+                .unwrap(),
+        )
+    }
+
+    /// Renames the gen outputs to be what rust and bindgen expect on linux distros
+    ///
+    /// IE halide.a -> libhalide.a
+    pub fn rename(&self) -> Result<()> {
+        println!(
+            "file name: {:?}",
+            self.rs_output
+                .join(self.gen_name.as_str())
+                .with_extension("a")
+                .to_str()
+                .unwrap()
+        );
+        println!(
+            "file out: {:?}",
+            self.rs_output
+                .join(String::from("lib").add(self.gen_name.as_str()))
+                .with_extension("a")
+                .to_str()
+                .unwrap()
+        );
+        fs::rename(
+            self.rs_output
+                .join(self.gen_name.as_str())
+                .with_extension("a")
+                .to_str()
+                .unwrap(),
+            self.rs_output
+                .join(String::from("lib").add(self.gen_name.as_str()))
+                .with_extension("a")
+                .to_str()
+                .unwrap(),
+        )
+    }
+
+    /// Run bindgen on a generator
+    ///
+    /// This bind only creates the genname funtion binding and specificly blocklists the buffer_t
+    pub fn bind(&self) -> Result<()> {
+        let bindings = bindgen::Builder::default()
+            .header(
+                self.rs_output
+                    .join(self.gen_name.as_str())
+                    .with_extension("h")
+                    .to_str()
+                    .unwrap(),
+            )
+            .allowlist_function(self.gen_name.as_str())
+            .blocklist_item("halide_buffer_t")
+            .generate()
+            .expect("unable to gen bindings")
```

Comment:
Prefer to propagate errors rather than panicing, or document why this panic is believed to be impossible.